File: src/main/java/com/balatro/structs/InstanceParams.java
package com.balatro.structs;

import com.balatro.enums.Deck;
import com.balatro.enums.Stake;
import com.balatro.enums.Voucher;
import org.jetbrains.annotations.NotNull;

public class InstanceParams {

    private Deck deck;
    private Stake stake;
    private boolean showman;
    public int sixesFactor;
    public long version;
    public int vouchers;

    public InstanceParams(Deck deck, Stake stake, boolean showman, long version) {
        this.deck = deck;
        this.stake = stake;
        this.showman = showman;
        this.sixesFactor = 1;
        this.version = version;
        this.vouchers = 0;
    }

    public Deck getDeck() {
        return deck;
    }

    public void setDeck(Deck deck) {
        this.deck = deck;
    }

    public Stake getStake() {
        return stake;
    }

    public void setStake(Stake stake) {
        this.stake = stake;
    }

    public boolean isShowman() {
        return showman;
    }

    public void setShowman(boolean showman) {
        this.showman = showman;
    }

    public int getSixesFactor() {
        return sixesFactor;
    }

    public void setSixesFactor(int sixesFactor) {
        this.sixesFactor = sixesFactor;
    }

    public long getVersion() {
        return version;
    }

    public void setVersion(long version) {
        this.version = version;
    }

    public void activateVoucher(@NotNull Voucher voucher) {
        vouchers = vouchers | (1 << voucher.ordinal());
    }

    public boolean isVoucherActive(@NotNull Voucher voucher) {
        return (vouchers & (1 << voucher.ordinal())) != 0;
    }
}

File: docs/js/immolate2/src/utils/InstanceParams.js
import DeckType from '../enums/DeckType.js';
import StakeType  from '../enums/Stake.js';
import Version  from '../enums/Version.js';

export class InstanceParams {
	constructor(d, s, showman, v) {
		this.deck = d || DeckType.RED_DECK;
		this.stake = s || StakeType.WHITE_STAKE
		this.showman = showman || false;
		this.sixesFactor = 1;
		this.version = v || Version.v_101c.getVersion();
		this.vouchers = new Set();
	}

	getDeck() {
		return this.deck;
	}

	setDeck(deck) {
		this.deck = deck;
	}

	getStake() {
		return this.stake;
	}

	setStake(stake) {
		this.stake = stake;
	}

	isShowman() {
		return this.showman;
	}

	setShowman(showman) {
		this.showman = showman;
	}

	getSixesFactor() {
		return this.sixesFactor;
	}

	setSixesFactor(sixesFactor) {
		this.sixesFactor = sixesFactor;
	}

	getVersion() {
		return this.version;
	}

	setVersion(version) {
		this.version = version;
	}

	getVouchers() {
		return this.vouchers;
	}

	setVouchers(vouchers) {
		this.vouchers = vouchers;
	}
}